module/item/systems/repair.js

```diff
@@ -150,6 +160,32 @@ class Repair {
             }
         }
 
+        if (gmRepair) {
+            await this.gmRepair(data)
+        } else {
+            await this.commonRepair(data, simulate)
+        }
+    }
+
+    attachHtmlListeners(html, data) {
+        html.find('.component-cost').on('change', this._onComponentCostChange.bind(this, data))
+    }
+
+    _onComponentCostChange(data, event) {
```

Comment:
Diagrams have a list of components and these components also have a price attached. See the link to the macro I posted

Reply:
Yeah, that's what I use for 'ownedComponents' which are in the actors inventory - see `RepairData.repairPrice`.  
These inputs are only for the missing components - I wanted to avoid the compendium lookups in this case since the compendium is not a hard dependency of the system. 
Also the system allows to create your own diagrams with any components - not necessarily from compendiums

Reply:
Could you add a check if the compendium is present and if not query the user for prices?